File: LeetCode/JavaScript/src/two_sum.js
// @author Anirudh Sharma

var twoSum = function (nums, target) {
    // Array to store the result
    result = [];
    // Map to store the difference and its index
    index_map = new Map();
    // Loop for each element in the array
    for (let i = 0; i < nums.length; i++) {
        let difference = target - nums[i];
        if (index_map.has(difference)) {
            result[0] = i;
            result[1] = index_map.get(difference);
            break;
        } else {
            index_map.set(nums[i], i);
        }
    }
    return result;
};

let nums = [2, 7, 11, 15];
let target = 9;
console.log(twoSum(nums, target));

nums = [3, 2, 4];
target = 6;
console.log(twoSum(nums, target));

File: LeetCode/Python/src/Two_Sum.py
"""
@author - Anirudh Sharma
"""
from typing import List


def twoSum(nums: List[int], target: int) -> List[int]:
    # List to store results
    result = []
    # Dictionary to store the difference and its index
    index_map = {}
    # Loop for each element
    for i, n in enumerate(nums):
        # Difference which needs to be checked
        difference = target - n
        if difference in index_map:
            result.append(i)
            result.append(index_map[difference])
            break
        else:
            index_map[n] = i
    return result


numbers = [2, 7, 11, 15]
target_value = 9
print(str(twoSum(numbers, target_value)))

numbers = [3, 2, 4]
target_value = 6
print(str(twoSum(numbers, target_value)))
